packages/matter.js/src/protocol/interaction/InteractionServer.ts

```diff
@@ -132,11 +146,26 @@ export function ClusterServer<F extends BitSchema, SF extends TypeFromBitSchema<
         acceptedCommandList: new Array<CommandId>(),
         generatedCommandList: new Array<CommandId>(),
     };
+
     const attributeList = new Array<AttributeId>();
     for (const attributeName in attributeDef) {
         const capitalizedAttributeName = capitalize(attributeName);
+        // logger.info(`check this for REQUIRED Attributes ${JSON.stringify(attributeName)}`)
+
+        const { requiredIf, optionalIf } = attributeDef[attributeName]
+        const requiredOk = validateInitialAttributesExist<F, SF, A, C, E>(requiredIf, clusterDef, attributesInitialValues, attributeName, supportedFeatures);
+        if(!requiredOk){
+            logger.warn(`*** InitialAttributeValue for "${clusterDef.name}/${attributeName}" is REQUIRED by supportedFeatures:${JSON.stringify(supportedFeatures)} but is not set`);
+        }
+        const optionalOk = validateInitialAttributesExist<F, SF, A, C, E>(optionalIf, clusterDef, attributesInitialValues, attributeName, supportedFeatures);
```

Comment:
why we need this? optionalis optional, so alles if not there

Reply:
I read the spec as optional being recommended for best support across all consumers.  So noting it for developers using the library is helpful.